test.py

```diff
@@ -0,0 +1,65 @@
+import os
+import pytest
+import subprocess
+import testinfra
+
+@pytest.fixture(scope='session')
+def host(request):
+    subprocess.check_call(['docker', 'build', '-t', 'radicale', '.'])
```

Comment:
The image name can be skipped, which can then avoid a conflict with a local `radicale` image, no?

In bash:
```
IMAGE_ID=$(docker build -q .)
docker run -it --rm "$IMAGE_ID"
```

Reply:
Ah well spotted. It can either be skipped, or in test I could call it something like `image-under-test`. 

Actually in order to tests to be self contained, probably best to do the latter otherwise you'd have to build radicale locally each time prior to running tests.